nixos/modules/services/web-servers/tomcat.nix

```diff
@@ -36,42 +32,76 @@ in
       };
 
       baseDir = mkOption {
+        type = lib.types.path;
         default = "/var/tomcat";
         description = "Location where Tomcat stores configuration files, webapplications and logfiles";
       };
 
+      logDirs = mkOption {
+        default = [];
+        type = types.listOf types.path;
+        description = "Directories to create in baseDir/logs/";
+      };
+
+      extraConfigFiles = mkOption {
+        default = [];
+        type = types.listOf types.path;
+        description = "Extra configuration files to pull into the tomcat conf directory";
+      };
+
+      environment = mkOption {
+        default = null;
+        example = "/etc/nixos/tomcat-environment";
```

Comment:
A file of environment variables is not the nicest interface. Instead it should be a list of strings:

```
extraEnvironment = mkOption {
   type = types.listOf types.str;
   # ...
};
# ...
systemd.servies.tomcat.serviceConfig = {
  Environment = [
    # ...
   ] ++ extraEnvironment;
};
```

if somebody needs a file, it is always possible to set this manually outside the module: `systemd.servies.tomcat.serviceConfig.EnvironmentFile`
Or use nix to split a file in lines..